_data/languages.yml

```diff
@@ -0,0 +1,4 @@
+- code: en
+  name: English
+- code: de
+  name: Deutsch
```

Comment:
When done testing, this should be removed. The German translation is nowhere near ready

Reply:
I actually would prefer the language menu `_includes/languages.html` use the `site.languages` array in the config file (instead of this `site.data.languages` array), and only take the language _names_ from elsewhere (e.g. this `languages.yml`). That'd make it less error-prone. Couldn't figure out a way (yet) though how to print value (languages name) from array y for each of the values in array x (site config).

Reply:
Post created on the/a Jekyll forum, hoping for an answer :) https://talk.jekyllrb.com/t/display-data-from-array-x-in-for-loop-based-on-array-y/6677

Reply:
My suggestion on the forums was make languages.yml a hash object not a list.

```yaml
de: Deutch
en: English
```

---

Using

```yaml
languages: ["en", "de"]
```

Or the neater

```yaml
languages:
  - en
  - de
```

You can build the menu with

```liquid
{% for lang_code in site.languages %}
  {% assign lang_name = site.data.languages[lang_code]
  <a href="/{{ lang_code }}/">
    {{ lang_name }}
  </a>
{% endfor %}
```

Reply:
OR you have to do a filter from the list with longer and less clear code.


```liquid
{% for lang_code in site.languages %}
  {% assign lang_map = site.data.languages | where: 'code', lang_code | first %}
  <a href="/{{ lang_code }}/">
    {{ lang_map.name }}
  </a>
{% endfor %}
```


Reference for `where`:

https://michaelcurrin.github.io/dev-cheatsheets/cheatsheets/jekyll/liquid/all-filters.html#array-filters

Reply:
Hi @MichaelCurrin, thanks again. In my forum post (reply to yours, which got automatically hidden), I wrote that I thought afterwards that languages should actually be sorted alphabetically - which can't be done (I think) when the forloop is based on `site.languages` (because the names aren't there). Instead I thought to keep `_data/languages.yml` an array with code and name, and have the following code to build the menu:
```
{% assign languages = site.data.languages | sort: 'name' %}
{% for language in languages %}
  {% if site.languages contains language.code %}
    <a class="dropdown-item" onchange="location = this.value;"
      href="{% if language.code != 'en' %}/{{ language.code }}{% endif %}{{ page.url | permalink: pretty }}">{{ language.name }}</a>
  {% endif %}
{% endfor %}
```
Do you think this makes sense?

Reply:
I would just manually keep the config languages sorted but yes if you want sort programmatically then this will work

I'd take out the permalink part as that doesn't do anything 

In your config you can set

```yaml
title: # ...
description: # ...
permalink: pretty 
```

And then page.url will be pretty already and the output files will be `/about/index.html/` rather than `/about.html`. So safer to do in another PR maybe.

Also if you use a subpath on gh pages then you need to add a prefix like this:

```
{{ page.url | relative_url }}
```

Reply:
Ahh, I see now about the `permalink: pretty` I was already confused :) It seems that the output file already is `/about/index.html` because `about.md` has `permalink: /about/`. But I guess changing that to `permalink: pretty` throughout the site is less error-prone. Will do that in another PR :)

> Also if you use a subpath on gh pages then you need to add a prefix like this:

The lang menu links build fine locally without `| relative_url` - are you saying that it builds slightly differently on GitHub and requires this to be added?

I saved your dev cheatsheet btw, thanks, very handy!

Reply:
Oh yeah you can change on a per file basis
```
---
title: About
permalink: /about/
---
```

If this doesn't get what you want in the config:

```yaml
permalink: pretty
```

Reply:
In this case you don't strictly need `relative_url`.

Since your repo is `antennapod.github.io` and the site is `https://antennapod.github.io/`. (And in this case you get that redirecting to the `.org` URL)

If your repo actually had a name like `my-repo`, then the URL for the site root would be `https://antennapod.github.io/my-repo/` 

And assets would be on that too.
e.g.

```
https://antennapod.github.io/my-repo/assets/main.js
```

So in that case, `{{ '/assets/main.js }}` would _break_.

While `{{ '/assets/main.js | relative_url }}` would turn into a valid URL s `/my-repo/assets/main.js`

Reply:
Thanks for the cheatsheet bookmark, and feel free to add issues or PRs.

Reply:
> If this doesn't get what you want in the config:
> `permalink: pretty`

It appears we already have `permalink: /:categories/:title`. At the same time we also have
```
collections:
  documentation:
    output: "true"
  blog:
    output: "true"
    permalink: /blog/:year/:month/:slug
```
I think the generic permalink setting is actually only relevant to the documentation, so I'll try moving it there, and making the generic permalink 'pretty'. But that's for a separate PR :)

Thanks again!